lib/net/modules/dns.toit

```diff
@@ -303,31 +380,53 @@ class DnsClient:
       // Ignore negative TTLs.
       ttl = max 0 ttl
 
+      min-ttl = min min-ttl ttl
+
       rd-length := BIG-ENDIAN.uint16 response position
       position += 2
-      if type == RECORD-A and query.accept-ipv4:
-        if rd-length != 4: throw (DnsException "Unexpected IP address length $rd-length")
-        if case-compare_ r-name q-name:
-          result := net.IpAddress
-              response.copy position position + 4
-          if ttl > 0:
-            trim-cache_ cache_
-            cache_[query.name] = CacheEntry_ result ttl
-          return result
+      if type == overall-type and case-compare_ r-name q-name:
+        if type == RECORD-A or type == RECORD-AAAA:
+          len := type == RECORD-A ? 4 : 16
+          if rd-length != len: throw (DnsException "Unexpected IP address length $rd-length")
+          result.add
+              net.IpAddress
+                  response.copy position position + len
+        else if type == RECORD-CNAME or type == RECORD-PTR:
+          result.add
+              decode-name response position: null
+        else if type == RECORD-TXT:
+          if response[position] != rd-length - 1: throw "WTF"
```

Comment:
Maybe improve the thrown exception here.

Reply:
Done.